flocker/cli/script.py

```diff
@@ -126,18 +115,59 @@ def __init__(self, ssh_configuration=None, ssh_port=22):
         self.ssh_configuration = ssh_configuration
         self.ssh_port = ssh_port
 
-    def _configure_ssh(self, deployment):
+    def main(self, reactor, options):
+        """
+        See :py:meth:`ICommandLineScript.main` for parameter documentation.
+
+        :return: A ``Deferred`` which fires when the deployment is complete or
+                 has encountered an error.
+        """
+        if options["nossh"]:
+            ready = succeed(None)
+        else:
+            ready = self._configure_ssh(options["deployment_config"].keys())
+
+        body = dumps({"applications": options["application_config"],
+                      "deployment": options["deployment_config"]})
+        ready.addCallback(
+            lambda _: post(options["url"], data=body,
```

Comment:
:question: 

Ideally I think this would also get the reactor as an argument.  On the other hand, since this is `flocker-deploy`, who cares.